### PR TITLE
Add exterior equipment

### PR DESCRIPTION
**Pull request by jmarrec**
Here's a little Xmas gift, courtesy of EffiBEM

The E+ `Exterior:Lights` was already wrapped in OS. I implemented the two others from E+: the `Exterior:FuelEquipmen`t and `Exterior:WaterEquipment`.

I created two abstract base classes:
* `ExteriorLoadDefinition`
* `ExteriorLoadInstance`

I modified `ExteriorLights` and `ExteriorLightsDefinition` to inherit from the base classes, and also added unit tests: model, FT AND RT.

I added 4 other classes, for which I wrote the model implementation, the forward translator, the reverse translator, and included unit tests for each (model/FT/RT)

* `ExteriorFuelEquipmentDefinition` and `ExteriorFuelEquipment`
* `ExteriorWaterEquipmentDefinition` and `ExteriorWaterEquipment`

Note: I didn't have time to add them to the OS App, nor to run the full suite of tests including regression tests (I ran all openstudio_model_tests and openstudio_energyplus_tests)

### CODE REVIEW

**Comment by nrel-bot-3** [comment]
@jmarrec @lgentile it has been 7 days since this pull request was last updated.

**Comment by nrel-bot-3** [comment]
@jmarrec @lgentile it has been 7 days since this pull request was last updated.

**Comment by nrel-bot-3** [comment]
@jmarrec @lgentile it has been 7 days since this pull request was last updated.

File: openstudiocore/src/model/ExteriorWaterEquipmentDefinition.cpp [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+  }
+
+  ExteriorWaterEquipmentDefinition_Impl::ExteriorWaterEquipmentDefinition_Impl(const ExteriorWaterEquipmentDefinition_Impl& other,
+                                                               Model_Impl* model,
+                                                               bool keepHandle)
+    : ExteriorLoadDefinition_Impl(other,model,keepHandle)
+  {}
+
+  // TODO: remove
+  const std::vector<std::string>& ExteriorWaterEquipmentDefinition_Impl::outputVariableNames() const
```
[the remark this answers is not in the document]
Comment (jmarrec) [reply openstudiocore/src/model/ExteriorWaterEquipmentDefinition.cpp]: @macumber
The variables are on the instance objects, not the definition.
See exteriorfuelequipment.cpp for example, you’ll see them here.

[the remark this answers is not in the document]
Comment (macumber) [reply openstudiocore/src/model/ExteriorWaterEquipmentDefinition.cpp]: Sure, I guess I was wondering why you would remove outputVariableNames (which would be hard to do) vs just making it return an empty vector?  I think the current implementation returning empty vector is fine right?

[the remark this answers is not in the document]
Comment (jmarrec) [reply openstudiocore/src/model/ExteriorWaterEquipmentDefinition.cpp]: The todo remove might just have been taken from another load definition file to be honest.
Also if I recall correctly this method is just overriding a base class virtual method, so overriding it to provide nothing might not be needed (don’t quote me on this, I haven’t double checked)

**Pull request merged**